website/src/error-documentation.mjs

```diff
@@ -84,6 +81,14 @@ markdownParser.renderer.rules = {
   },
 };
 
+function codeHasBOM(codeHTML) {
+  const dom = new jsdom.JSDOM(codeHTML);
+  const mark = dom.window.document.querySelector("mark");
+  return (
+    codeHTML.startsWith("\u{feff}") || mark?.textContent.startsWith("\u{feff}")
```

Comment:
Blocking: This code is incorrect for the following HTML (assuming `ZWNBS` is U+FEFF):
```html
hello<mark>ZWNBSworld</mark>
```
This function should return `false` given the above HTML, but it will return `true`.

Reply:
Some test cases:
```html
<!-- has BOM -->
<mark>ZWNBShello</mark>
<mark data-code="E123">ZWNBShello</mark>
ZWNBShello

<!-- has no BOM -->
hello<mark>ZWNBSworld</mark>
helloZWNBSworld
```

Reply:
> Some test cases:
> 
> ```
> <!-- has BOM -->
> <mark>ZWNBShello</mark>
> <mark data-code="E123">ZWNBShello</mark>
> ZWNBShello
> 
> <!-- has no BOM -->
> hello<mark>ZWNBSworld</mark>
> helloZWNBSworld
> ```

I write some tests with this cases and if I did right, this possibilities are reached.